src/database.js

```diff
@@ -1,27 +1,28 @@
 const MONGODB_URI = process.env.MONGODB_URI || process.env.OPENSHIFT_MONGODB_DB_URL;
 const mongoose = require('mongoose');
 const bcrypt = require('bcryptjs');
+const { isBlank } = require('underscore.string');
 
 mongoose.Promise = Promise;
 mongoose.connect(MONGODB_URI);
 
 var Schema = mongoose.Schema;
 
 var UserSchema = new Schema({
-  name: String,
-  email: { type: String, unique: true },
-  username: { type: String, unique: true },
-  password: String,
+  name: { type: String, required: true },
+  email: { type: String, unique: true, required: true },
+  username: { type: String, unique: true, required: true },
+  password: { type: String, required: true },
```

Comment:
All four of these validations are very weak and will introduce regressions. I think this is where you need to use the `validator` property to, say, use `isEmail` and `isBlank`, etc.